### ISSUE TITLE
[Integration] - Unrecognised TS0601 Temperature/Humidity sensor

### ISSUE THREAD

**Issue by stevedh42** [sasu-drooz/Domoticz-Zigate#1515]
* Product Name : Tenky Tuya ZigBee Temperature Humidity Sensor
* Brand : Tenky
* URL on a web site : https://www.aliexpress.com/item/1005005012094422.html
* What do you expect : To be able to get temperature and humidity values from it.

* ZigBeeforDomoticZ version : 6.3.010

*Other info : This is a cheap Chinese zigbee temperature sensor with a display bought from aliexpress. Before I updated to 6.3.010 it was detected but didn't work.. old log also included just in case it is of any use.

* The logs from the pairing process
* export.json file as described here above
[pairing.log](https://github.com/zigbeefordomoticz/Domoticz-Zigbee/files/10561319/pairing.log)
[export.json.txt](https://github.com/zigbeefordomoticz/Domoticz-Zigbee/files/10561325/export.json.txt)
[pairing_old_6_3_005.log](https://github.com/zigbeefordomoticz/Domoticz-Zigbee/files/10561438/pairing_old_6_3_005.log)

**Comment by stevedh42** [comment]
I did try just adding 
    "_TZE200_whkgqxse"
to
TUYA_TEMP_HUMI = (
in tuyaConst.py

it was detected, but readings were zero, and got this in the log.

2023-02-01 21:07:17.107 Error: Sonoff Zigbee: Aberrant Temp: 1.8180266169757779e+78 (below -50 or above 100) for device: 8a34
2023-02-01 21:07:18.089 Error: Sonoff Zigbee: Aberrant Temp: 1.8180266169757779e+78 (below -50 or above 100) for device: 8a34

**Comment by pipiche38** [comment]
Which version of the plugin are you using?Envoyé de mon iPhoneLe 1 févr. 2023 à 22:08, stevedh42 ***@***.***> a écrit :﻿
I did try just adding
"_TZE200_whkgqxse"
to
TUYA_TEMP_HUMI = (
in tuyaConst.py
it was detected, but readings were zero, and got this in the log.
2023-02-01 21:07:17.107 Error: Sonoff Zigbee: Aberrant Temp: 1.8180266169757779e+78 (below -50 or above 100) for device: 8a34
2023-02-01 21:07:18.089 Error: Sonoff Zigbee: Aberrant Temp: 1.8180266169757779e+78 (below -50 or above 100) for device: 8a34

**Comment by pipiche38** [comment]
could you please go to the hotfix-6.3.011 branch

```
git pull
git checkout hotfix-6.3.011
```

remove the device from domoticz et the plugin
restart the plugin

enable debug via the 'WebUI' > Debug

Input
Tuya

<img width="1670" alt="Screenshot 2023-02-02 at 08 40 41" src="https://user-images.githubusercontent.com/8291674/216261548-e5443f8d-b2ad-4040-8870-8f3f818bb948.png">

Then do the pairing, leave it like that during 5 or 10 minutes

finaly drop me the log file 
- Log/PluginZigbee-xx.log
and the export.json 'Plugin Device List RAW'

**Comment by stevedh42** [comment]
thanks
Please find attached log and raw plugin list. Hope these help. From googling looks like this device is usually described as Temperature/Humidity sensor with clock... although I think the clock is unreadable.


[PluginZigbee_04.log](https://github.com/zigbeefordomoticz/Domoticz-Zigbee/files/10567277/PluginZigbee_04.log)
[export2.json.txt](https://github.com/zigbeefordomoticz/Domoticz-Zigbee/files/10567286/export2.json.txt)

**Comment by pipiche38** [comment]
The data transmitted looks totally un-expected.

Is this values correct ?

Temp: 0xc2 -> 19.4
Humi: 0x37 -> 55
Battery: 0x64 -> 100
0a: 0x68 -> 104 ???
0b: 0xc8 -> 200 ???

**Comment by stevedh42** [comment]
Yes those values look correct, I'm pretty sure it was around 19 degrees at that time..
Be interesting to know what those other values are, pity I didn't set the clock correctly as maybe they might have had something to do with the time..

**Comment by pipiche38** [comment]
I beleive you can set some Alarm, so it might be related to a Temp Min / Max for example

Can you switch to a new branch (untested)

```
git pull
git checkout tuya-cluster/recursive-scan
git pull
```

simply restart the plugin

and let see . Please cross check if any errors can occures in the Domoticz Logs and the plugin logs
leave the debug flag as they are

**Comment by stevedh42** [comment]
On the plus side... it set the date and time automatically, which surprised me.

On the minus side It is still reading the data incorrectly, I checked and as far as I can tell I updated correctly.

2023-02-02 18:10:38.540 Error: Sonoff Zigbee: Aberrant Temp: 2.0496107954504102e+78 (below -50 or above 100) for device: db4f
2023-02-02 18:10:42.503 Error: Sonoff Zigbee: Aberrant Temp: 2.0496107954504102e+78 (below -50 or above 100) for device: db4f

BTW current temperature is 17.4, although it could have been 17.6 and Humidity is 51

**Comment by stevedh42** [comment]
Log just in case it is helpful
[PluginZigbee_04.log](https://github.com/zigbeefordomoticz/Domoticz-Zigbee/files/10572007/PluginZigbee_04.log)

**Comment by stevedh42** [comment]
If you want me to check a file or something I can.

**Comment by pipiche38** [comment]
Please provides the output of command

git status

**Comment by stevedh42** [comment]
steve@ubuntu1:~/domoticz/plugins/Domoticz-Zigbee$ sudo git status
On branch tuya-cluster/recursive-scan
Your branch is up to date with 'origin/tuya-cluster/recursive-scan'.

nothing to commit, working tree clean

**Comment by pipiche38** [comment]
kindly retrieve the latest update on the same branch and restart the plugin

**Comment by stevedh42** [comment]
Done.. 
still has same error but thanks for looking at this.

2023-02-03 09:49:35.108 Error: Sonoff Zigbee: Aberrant Temp: 2.2464573471538477e+78 (below -50 or above 100) for device: db4f

I've attached log.
[PluginZigbee_04.log](https://github.com/zigbeefordomoticz/Domoticz-Zigbee/files/10577808/PluginZigbee_04.log)

BTW the screen of db4f is showing (sorry to complicate things there are 2 of these connected to zigbee, but only 1 I can look at at the moment)
temp  19.4
humidity 57%


And Git status

On branch tuya-cluster/recursive-scan
Your branch is up to date with 'origin/tuya-cluster/recursive-scan'.

nothing to commit, working tree clean

**Comment by pipiche38** [comment]
not sure I understand, is the temp/humi correctly reported ?

**Comment by stevedh42** [comment]
No sorry, still getting 0 for temp and 50 for humid which never change.
still getting this error in the log.
2023-02-03 09:49:35.108 Error: Sonoff Zigbee: Aberrant Temp: 2.2464573471538477e+78 (below -50 or above 100) for device: db4f

I just included the temperature read directly from the display of the device incase that helped.

**Comment by pipiche38** [comment]
Ok, try again

git pull (on the same branch)

**Comment by stevedh42** [comment]
no :(

2023-02-03 11:31:34.981 Error: Sonoff Zigbee: Aberrant Temp: 2.3043533917725058e+78 (below -50 or above 100) for device: db4f
2

Log attached

[PluginZigbee_04.log](https://github.com/zigbeefordomoticz/Domoticz-Zigbee/files/10578461/PluginZigbee_04.log)

**Comment by pipiche38** [comment]
No clue what is happening, just added some debug , can you update and restart

**Comment by stevedh42** [comment]
Done.. log attached
[PluginZigbee_04.log](https://github.com/zigbeefordomoticz/Domoticz-Zigbee/files/10578809/PluginZigbee_04.log)

**Comment by pipiche38** [comment]
once more ;-)

**Comment by stevedh42** [comment]
ALMOST :) I was watching domoticz log waiting for the error to pop up and it didn't

Temperature is correct but humidity is wrong

Temp + Humidity	THGN122/123/132, THGR122/228/238/268	20.8 C, 5 %

Humidity should be 56%

log attached.



[PluginZigbee_04.log](https://github.com/zigbeefordomoticz/Domoticz-Zigbee/files/10578999/PluginZigbee_04.log)

**Comment by pipiche38** [comment]
please switch back to hotfix-6.3.011

```
git pull
git checkout hotfix-6.3.011
```

and then restart the plugin

**Comment by stevedh42** [comment]
Done thanks,
Humidity still show 5% instead of now 55%

Log attached

[PluginZigbee_04.log](https://github.com/zigbeefordomoticz/Domoticz-Zigbee/files/10579322/PluginZigbee_04.log)
[PluginZigbee_log_error_history.json.txt](https://github.com/zigbeefordomoticz/Domoticz-Zigbee/files/10579323/PluginZigbee_log_error_history.json.txt)

**Comment by pipiche38** [comment]
indeed they were an error, try again

**Comment by stevedh42** [comment]
YAY!!
all appears to be working, thanks, I'm guessing this patch was a bit more complicated then normal.

**Comment by pipiche38** [comment]
Good, so it will merged on stable6 at a point of time.

**Comment by pipiche38** [comment]
Can you kindly switch to the branch config-release-6.3.011

```
git pull
git checkout config-release-6.3.011

```
and redo the all test from pairing

**Comment by stevedh42** [comment]
everything looks good.

Did the pull and checkout.
removed it from Domoticz and the plugin
rebooted plugin.
repaired it.
everything working well. (also confirmed battery level by using uncharged battery)

**Comment by pipiche38** [comment]
Ok, thanks for the cross-check.
You can remain on that branch until we published release 6.3.011

thanks for your support
